lib/zlib.js

```diff
@@ -706,57 +706,64 @@ Zlib.prototype.params = function params(level, strategy, callback) {
 // generic zlib
 // minimal 2-byte header
 function Deflate(opts) {
-  if (!(this instanceof Deflate))
+  if (!new.target) {
```

Comment:
```suggestion
  if (new.target !== Deflate) {
```
if you also want to normalize subclasses